Reply/app/src/main/java/com/materialstudies/reply/ui/compose/ComposeFragment.kt

```diff
@@ -137,92 +124,76 @@ class ComposeFragment : Fragment() {
-            // shared element transition.
-            startView = requireActivity().findViewById(R.id.fab)
-            endView = binding.emailCardView
-            duration = resources.getInteger(R.integer.reply_motion_duration_large).toLong()
-            interpolator = requireContext().themeInterpolator(R.attr.motionInterpolatorPersistent)
-        }
-        returnTransition = Slide().apply {
-            duration = resources.getInteger(R.integer.reply_motion_duration_medium).toLong()
-            interpolator = requireContext().themeInterpolator(R.attr.motionInterpolatorOutgoing)
-        }
-        startPostponedEnterTransition()
-    }
-
-    private fun containerTransformRecipientChipToCard() {
+    private fun expandChip(chip: View) {
         val transform = MaterialContainerTransform().apply{
-            // Manually tell the container transform which Views to transform between.
-            startView = binding.recipientChipGroup
+            startView = chip
             endView = binding.recipientCardView
-
-            // Optionally add a curved path to the transform
-            // pathMotion = MaterialArcMotion()
-
-            // Since View to View transforms often are not transforming into full screens,
-            // remove the transition's scrim.
             scrimColor = Color.TRANSPARENT
+            // Have the transform match the endView card's native elevation as closely as possible.
+            endElevation = requireContext().resources.getDimension(
+                R.dimen.email_recipient_card_popup_elevation_compat
+            )
+            // Avoid having this transform from running on both the start and end views by setting
+            // its target to the endView.
+            addTarget(binding.recipientCardView)
+        }
 
+        // Configure the analogous collapse transform back to the recipient chip. This should
+        // happen when the card is clicked, any region outside of the card (the card's transparent
+        // scrim) is clicked, or when the back button is pressed.
+        binding.run {
+            recipientCardView.setOnClickListener { collapseChip(chip) }
+            recipientCardScrim.setOnClickListener { collapseChip(chip) }
         }
-        // Begin the transition by changing properties on the start and end views or
-        // removing/adding them from the hierarchy.
+        closeRecipientCardOnBackPressed.expandedChip = chip
+        closeRecipientCardOnBackPressed.isEnabled = true
+
         TransitionManager.beginDelayedTransition(binding.composeConstraintLayout, transform)
         binding.recipientCardView.visibility = View.VISIBLE
-        binding.recipientChipGroup.visibility = View.GONE
-
+        // Using INVISIBLE instead of GONE ensures the chip's parent layout won't shift during
+        // the transition due to chips being effectively removed.
+        chip.visibility = View.INVISIBLE
```

Comment:
insterestinggg, so the chip is technically still clickable, but is blocked by the scrim?

Reply:
Yea.
We could set the chip to `clickable = false`, but I'm not sure it's necessary

Reply:
yeah i think this is fine as is